components/eam/src/chemistry/utils/mam_support.F90

```diff
@@ -32,14 +35,26 @@ subroutine assign_la_lc( imode,      ispec,          & ! in
                                                ! 0 = number concentration
                                                ! other = mass concentration
    integer, intent(out)    :: la               ! index of interstitial aerosol
-   integer, intent(out)    :: lc               ! index of cloudborne aerosol (la+ pcnst)
+   integer, intent(out)    :: lc               ! index of cloudborne aerosol
+   logical, optional, intent(in) :: is_lc_append_in  ! if cloudborne aerosol is appended after interstitial aerosols
+
+   logical :: is_lc_append   
+
+   ! the default option is treat cloudborne aerosols in separated array
+   is_lc_append = .false.
+   if (present(is_lc_append_in)) is_lc_append=is_lc_append_in
 
    if (ispec == 0) then
```

Comment:
This looks good but a slightly better way would be to make `ispec` an optional variable as well. Here instead of `ispec == 0`, we can check if it is `present` or not.  This will enable this subroutine to be invoked using minimum input from the user (for example, one can skip `ispec` as an argument for `num` index).

It is fine as is as well and we can merge it.